rxjava/src/main/java/com/linecorp/armeria/rxjava/RequestContextScalarSupplierMaybe.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.rxjava;
+
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.util.SafeCloseable;
+
+import io.reactivex.rxjava3.core.Maybe;
+import io.reactivex.rxjava3.core.MaybeObserver;
+import io.reactivex.rxjava3.core.MaybeSource;
+import io.reactivex.rxjava3.internal.fuseable.ScalarSupplier;
+
+final class RequestContextScalarSupplierMaybe<T> extends Maybe<T>
+        implements ScalarSupplier<T> {
+    private final MaybeSource<T> source;
+    private final RequestContext assemblyContext;
+
+    RequestContextScalarSupplierMaybe(MaybeSource<T> source, RequestContext assemblyContext) {
+        this.source = source;
+        this.assemblyContext = assemblyContext;
+    }
+
+    @Override
+    protected void subscribeActual(MaybeObserver<? super T> s) {
+        try (SafeCloseable ignored = assemblyContext.push()) {
+            source.subscribe(new RequestContextMaybeObserver<>(s, assemblyContext));
+        }
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public T get() {
+        try (SafeCloseable ignored = assemblyContext.push()) {
+            return ((ScalarSupplier<T>) source).get();
+        }
+    }
```

Comment:
The value of `ScalarSupplier`, which was `ScalarCallable` in RxJava 2, was captured when calling `Maybe.just(T)`.
I wonder do we need to push `RequestContext` here? If not, I think we could remove all `RequestContext*ScalarSupplier*". 🧐

Reply:
I think we don't have to push the context here. 😄

Reply:
On second thought, I think there a chance that a user specifies a class that extends `Maybe` and implements `ScalarSupplier`.
The specified class might need the context in `get()` method, so I think we need this.

Reply:
Hmm... `ScalarSupplier` is the same as `Supplier` except it abandon `throws Throwable`. If a user trigger effects in `get()` method, it seems a wrong usage of `ScalarSupplier`. Because it should hold const and scalar value. 
https://github.com/ReactiveX/RxJava/blob/3.x/src/main/java/io/reactivex/rxjava3/internal/fuseable/ScalarSupplier.java#L19 
I think the user has to use `Supplier` for generating value dynamically.

Reply:
Ah, yes. That's a good point. 👍